SA3/qa_web/views.py

```diff
@@ -29,6 +29,41 @@ def index(request):
     return render(request, 'qa_web/index.html', context={'questions': Questions.objects.all()})
 
 
+@login_required(login_url='/login/')
+def edit_profile(request):
+    if request.method=='GET':
+        form = UserProfile()
+        # return render_to_response('qa_web/UserProfile.html', RequestContext(request, {'form': form, }))
+        return render(request,'qa_web/EditUserProfile.html', context={'form': form})
```

Comment:
Good consistence throughout the website to generate the form on GET, and validating/populating the database on POST.

Reply:
thank you!